datasets/youtube_caption_corrections/youtube_caption_corrections.py

```diff
@@ -0,0 +1,117 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Dataset built from <auto-generated, manually corrected> caption pairs of
+YouTube videos with labels capturing the differences between the two."""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+
+import datasets
+
+
+_CITATION = """\
+url={https://github.com/2dot71mily/youtube_captions_corrections}
+},
+year={2020}
+}
+"""
+
+_DESCRIPTION = """\
+Dataset built from pairs of YouTube captions where both an `auto-generated` and
+a `manually corrected` caption was available for a single specified language.
+This dataset looks only at simple (e.g. ignoring case/punctuation/stop words)
+two-way (e.g. ignoring single-sided insertions) single-token differences between the
+auto-gen and manual transcripts.
+More complex differences in the original repo are not captured here.
+"""
+
+_LICENSE = "MIT License"
+
+_CHANNEL_NAMES = [
+    "3Blue1Brown",
+    "Aurélien_Géron",
+    "Jeremy_Howard",
+    "Luis_Serrano",
+    "TED",
+    "Weights_&_Biases",
+    "nature_video",
+    "Alfredo_Canziani",
+    "DeepMind",
+    "Pieter_Abbeel",
+    "Veritasium",
+    "minutephysics",
+    "stanfordonline",
+]
```

Comment:
good content here :)

Reply:
I had huggingface on the list, but the youtube channel doesn't seem to have any auto-gen, manually-corrected caption pairs, yet! However, I plan to keep occasionally rerunning my scripts to update the content per channel, and add more channels, too.

For this reason, I was originally hoping I could avoid hardcoding these channel names, and instead grab all jsons in the github repo dir.

I see you guys use the requests library to download the URLS, but I couldn't find a util in the code that would grab all links from a webpage request. Is that already out there, or would it be of interest, or not appropriate for this library?

Reply:
What we use is the `dl_manager` download methods. 
Basically what you can do is have one json in your repo with a list of all available channels.
Then in the script you can first download this json to get the list of channels, and them iterate over the channel names to download their json files.

Also you mentioned that you plan to re-run you script and add more channels.
That's a wonderful idea :) 
One suggestions I may add is have some some of versioning on your repo that we can use in the dataset script to specify which version we want to load (by date or using a specific github release tag for example). You want organize your versioning as you want. Either having one folder per date in the repo, or doing github releases for example.
What do you think ?